### DISCUSSION TITLE
DEVC  is not within any mesh - Measuring HEATFLUX in a volume specified by XB

### DISCUSSION THREAD

**Discussion by ramsampath** [firemodels/fds#13860] [Q&A] [answered]
Hello,

I would like to output a max value of heatflux in a volume specified by XB .... where some of that volume intersects or overlaps a solid surface. 

The FDS manual says the following. 

_"...
&DEVC XB=..., QUANTITY='TEMPERATURE', ID='maxT', SPATIAL_STATISTIC='MAX' /
causes FDS to write out the maximum gas phase temperature over the volume bounded by XB. Other
SPATIAL_STATISTIC’s are discussed below. Some are appropriate for gas phase output quantities, some
for solid phase, and some for both. Note that if XB is used for a point device without a SPATIAL_STATISTIC,
then FDS will define XYZ to be the center of the volume defined by XB.
"_

In my case 

 However when I run fds I get the following warnings and the HEAT_FLUX is not output. 
 
 _DEVC A1_HEATFLUX is not within any mesh_

 Could you guide me on how to get the max heat flux among in a volume specified by XB where XB overlaps the solid ?

and also similarly WALL_TEMPERATURE.

 Attached is the sample FDS file (as txt) and I get the following warnings and don't get the heat flux value output. 
 Would really appreciate some help on this.

Thank you
Ram.
[FSource_1_FDS.txt](https://github.com/user-attachments/files/18056967/FSource_1_FDS.txt)

**Comment by drjfloyd** [comment]
&DEVC ID='A1_HEATFLUX', XB=3.0273, 3.2265, 2.0669, 2.094, 2.1115, QUANTITY='NET HEAT FLUX' SPATIAL_STATISTIC='MAX'  /

XB need six values.

**Comment by ramsampath** [comment]
Apologies for the overlook, it was a syntax error on my part @drjfloyd  thank you for pointing out.   The correct values are as follows and attached in the following fds file. I got past that error . 

&DEVC ID='A1_HEATFLUX', XB=3.0273, 3.2265, 2.0669, 2.0904, 2.094, 2.1115, QUANTITY='NET HEAT FLUX' SPATIAL_STATISTIC='MAX' /

Based on a visual overlook it should overlap some solid cells.
 - However this time it gives me values for MAX Heat flux as -10000000000.   Wondering what I am doing wrong, maybe I overlooked another detail . Would really appreciate help on this. 
 
[FSource_1_FDS.txt](https://github.com/user-attachments/files/18063463/FSource_1_FDS.txt)

**Reply by mcgratta** [reply]
I will run this case and check it out.

**Reply by mcgratta** [reply]
I ran the case with the latest source code and I do not see this large negative value. What version of FDS are you using?

**Reply by ramsampath** [reply]
Thank you @mcgratta . The version of FDS I am using is 6.9 
Fire Dynamics Simulator

 Current Date     : December  9, 2024  09:14:16
 Revision         : FDS-6.9.1-0-g889da6a-release
 Revision Date    : Sun Apr 7 17:05:06 2024 -0400
 Compiler         : Intel(R) Fortran Intel(R) 64 Compiler Classic for applications running on Intel(R) 64, Version 2021.9.0 Build 20230302_000000
 Compilation Date : Tue 04/09/2024  01:25 PM

**Comment by drjfloyd** [comment]
6.9.1 release:

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/FLOYDJ~1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/FLOYDJ~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">


s | C | kW/m2 | C | C | kW/m2
-- | -- | -- | -- | -- | --
Time | A1_TEMP | A1_HEATFLUX | DCP-A-EC_TEMP | DCP-A-EC_WALLTEMP | DCP-A-EC_HEATFLUX
0 | 20 | -10000000000 | 20 | -10000000000 | -10000000000



</body>

</html>

**Comment by drjfloyd** [comment]
I haven't looked to see if the DEVC actually includes any wall cells but if it does, then with the current source the DCP-A-EC_WALLTMEP doesn't look right:

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/FLOYDJ~1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/FLOYDJ~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">


s | C | kW/m2 | C | C | kW/m2
-- | -- | -- | -- | -- | --
Time | A1_TEMP | A1_HEATFLUX | DCP-A-EC_TEMP | DCP-A-EC_WALLTEMP | DCP-A-EC_HEATFLUX
0 | 20 | 0 | 20 | 0 | 0



</body>

</html>

**Reply by ramsampath** [reply]
@drjfloyd  - are the second set of values with a different version of fds ? I was getting the big negative numbers with FDS 6.9 . 

Also, attached is a representation of the volume 

which defines the XB of DEVC device. It seems to be just bordering the face of the OBST. Perhaps that's the issue ? Does it need to penetrate the OBST in order to correctly measure ? If so I can increase the bounds of the XB.

Thank you.
<img width="572" alt="2" src="https://github.com/user-attachments/assets/a24f4c18-07c7-499e-b1fc-3b5857f62fd4">

**Reply by drjfloyd** [reply, marked as answer]
As noted in my post the second set was "the current source"

It is best if the XB slightly overlaps the wall cells. As noted in the User's Guide

"You can further limit the search by using the sextuplet of coordinates XB to force FDS to only compute statistics for
surface cells within the given volume. Be careful to account for the fact that the solid surface might shift
to conform to the underlying numerical grid."

**Reply by mcgratta** [reply]
I'm going to see if I can create an `ERROR` if there are no values found within the search volume. That is what is happening here.

**Reply by ramsampath** [reply]
Thank you Jason and Kevin - @drjfloyd  and @mcgratta  - let me try to reposition the bounds so that they overlap.

**Reply by ramsampath** [reply]
Thank you much. I repositioned the bounds to make sure that they overlap in the following fds file and after a couple of adjustments I was able to get values for both WALL_TEMPERATURE and HEAT_FLUX . 

This specific case had this geometry bound which was attached to the device exactly lined up with the grid ? and when i placed the device on the boundary of that wall it fell on the adjacent grid cell, so this may have been the reason it wasn't finding the surface ?

Anyway, I was able get values after making sure there was considerable overlap.

Thank you again.

**Reply by mcgratta** [reply]
I am working to check to ensure that there is at least one point included in any `SPATIAL_STATISTIC` evaluation. FDS includes surface cells whose centers fall within the `XB` volume. If it is exactly at the boundary of the volume, there is a chance that it might not be included. Best to make the volume a bit bigger to make sure you get at least one point.

**Reply by ramsampath** [reply]
Thank you much @mcgratta  - that was indeed the case in this. Yes that  would greatly help troubleshoot, if there's some warning message printed if there are no surfaces in the volume .
